src/cargo/ops/cargo_test.rs

```diff
@@ -83,7 +84,12 @@ fn run_unit_tests(
     for &(ref pkg, ref kind, ref test, ref exe) in &compilation.tests {
         let exe_display = exe.strip_prefix(cwd).unwrap_or(exe).display();
         let mut cmd = compilation.target_process(exe, pkg)?;
+        let targets = pkg.manifest().targets();
+        let harness = targets.iter().any(|t| t.kind() == &TargetKind::Test && t.harness());
```

Comment:
I don't think this is the correct way to check for a harness.  For example, this won't work if the target is a lib or bin. It also won't work properly if some targets of a harness and some don't.   I think maybe the entire Target struct will need to be included in `Compilation::tests`?

Reply:
Makes sense. Will do. Thank you!